entity.go

```diff
@@ -82,3 +82,24 @@ func (e *Entity) ComponentFast(c Component) interface{} {
 func (e *Entity) ID() string {
 	return e.id
 }
+
+// Duplicate populates an array with the specified amounts of clones from an Entity.
+// NOTE: If you provide your components as pointers, the pointer will be duplicated.
+// Hence, the entities will "share" a component.
```

Comment:
I think so?

Cloning each component with reflect is not very practical IMO. Just feels a bit too hacky.

One potential option would be to add a `Clone() Component` function? And then this could return the component without a pointer?